### PR TITLE
Remove unused legacy-diagnostics-system type

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/VisualStudio/Core/Def/TableDataSource/AbstractTableEntriesSnapshot.cs [hunk 1]
```diff
@@ -1,303 +0,0 @@
...
-using Microsoft.VisualStudio.Shell.TableManager;
-using Microsoft.VisualStudio.Text;
-using Roslyn.Utilities;
-
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource;
-
-/// <summary>
-/// Base implementation of ITableEntriesSnapshot
-/// </summary>
-internal abstract class AbstractTableEntriesSnapshot<TItem> : ITableEntriesSnapshot
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TableDataSource/AbstractTableEntriesSnapshot.cs:27]: this abstract class had no subclasses.  So you can never have an instance of this class.  Deleting outright.  The remainder is fallout of htat.

File: src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs [hunk 2]
```diff
@@ -200,16 +182,6 @@ private static bool IsNonRoslynEntrySupportingSuppressionState(ITableEntryHandle
...
         => entry != null && !SuppressionHelpers.IsNotConfigurableDiagnostic(entry);
 
-    private static AbstractTableEntriesSnapshot<DiagnosticTableItem>? GetEntriesSnapshot(ITableEntryHandle entryHandle, out int index)
-    {
-        if (!entryHandle.TryGetSnapshot(out var snapshot, out index))
-        {
-            return null;
-        }
-
-        return snapshot as AbstractTableEntriesSnapshot<DiagnosticTableItem>;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs:210]: this 'as' must fail, as there are no actual instances of this type.  So thsi method collapsed to `return null;` which i then inlined to callsites.

File: src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs [hunk 3]
```diff
@@ -159,27 +159,9 @@ private static bool EntrySupportsSuppressionState(ITableEntryHandle entryHandle,
         isNoLocationDiagnosticEntry = !entryHandle.TryGetValue(StandardTableColumnDefinitions.DocumentName, out string filePath) ||
             string.IsNullOrEmpty(filePath);
 
-        var roslynSnapshot = GetEntriesSnapshot(entryHandle, out var index);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs:162]: see comment on GetEntriesSnapshot.  roslynSnapshot will always be null.  so the first if is taken unilaterally.

File: src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs [hunk 4]
```diff
@@ -228,12 +200,8 @@ public async Task<ImmutableArray<DiagnosticData>> GetSelectedItemsAsync(bool isA
             cancellationToken.ThrowIfCancellationRequested();
 
             DiagnosticData? diagnosticData = null;
-            var roslynSnapshot = GetEntriesSnapshot(entryHandle, out var index);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs:231]: same.  this is always null.  so we know we must take the else branch.

File: src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs [hunk 5]
```diff
@@ -234,20 +234,9 @@ async System.Threading.Tasks.Task<Solution> ConfigureSeverityAsync(CancellationT
...
-    private DiagnosticData? TryGetSingleSelectedEntry()
+    private static DiagnosticData? TryGetSingleSelectedEntry()
     {
-        if (_tableControl?.SelectedEntries.Count() != 1)
-        {
-            return null;
-        }
-
-        if (!_tableControl.SelectedEntry.TryGetSnapshot(out var snapshot, out var index) ||
-            snapshot is not AbstractTableEntriesSnapshot<DiagnosticTableItem> roslynSnapshot)
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs:245]: the 'or' always applies here.  so this method always returns null.

Comment (ToddGrun) [reply src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs:245]: Why not nuke this method then?

**Review by ToddGrun — APPROVED** [review]

File: src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs:245]: just making safe cut off points.  tehre are more followups.

**Pull request merged**